allennlp/modules/token_embedders/basic_token_embedder.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict
+
+from overrides import overrides
+import torch
+
+from allennlp.common import Params
+from allennlp.data import Vocabulary
+from allennlp.experiments import Registry
+from allennlp.modules import TokenEmbedder, TokenVectorizer
+
+
+@Registry.register_token_embedder("basic")
+class BasicTokenEmbedder(TokenEmbedder):
+    """
+    This is a ``TokenEmbedder`` that wraps a collection of :class:`TokenVectorizer` objects.  Each
+    ``TokenVectorizer`` embeds or encodes the representation output from one
+    :class:`~allennlp.data.TokenIndexer`.  As the data produced by a
+    :class:`~allennlp.data.fields.TextField` is a dictionary mapping names to these
+    representations, we take ``TokenVectorizers`` with corresponding names.  Each
+    ``TokenVectorizer`` embeds its input, and the result is concatenated in an arbitrary order.
+    """
+    def __init__(self, token_vectorizers: Dict[str, TokenVectorizer]) -> None:
+        super(BasicTokenEmbedder, self).__init__()
+        self._token_vectorizers = token_vectorizers
+
+    @overrides
+    def get_output_dim(self) -> int:
+        output_dim = 0
+        for vectorizer in self._token_vectorizers.values():
+            output_dim += vectorizer.get_output_dim()
+        return output_dim
+
+    def forward(self, text_field_input: Dict[str, torch.Tensor]) -> torch.Tensor:
+        assert self._token_vectorizers.keys() == text_field_input.keys(), "mismatched token keys"
```

Comment:
Because I was operating under the mistaken assumption that `assert` would show the failed assertion as well as the error message, so I wouldn't have to print out the mismatched keys myself.  Turns out I was wrong.  I'll switch it.